src/udev/udev-manager.c

```diff
@@ -697,6 +710,9 @@ static int event_queue_start(Manager *manager) {
 
         manager_reload(manager, /* force = */ false);
 
+        if (!manager_can_process_event(manager))
+                return 0;
```

Comment:
Hmm why not do this check as the first thing in the for loop? Then you don't have to duplicate it at the end of the for loop.

Reply:
As the result of manager_can_process_event() can be changed only after reloading or starting an event. Checking that in the beginning of a loop may be a waste of time. I added short comment for each call in the code.